pgmpy/models/FunctionalBayesianNetwork.py

```diff
@@ -191,3 +192,156 @@ def simulate(self, n_samples=1000, seed=None):
+        ...    return dist.Normal(mu, sigma)
+
+        >>> def x2_prior(parent):
+        ...    mu = pyro.sample("x2_mu", dist.Normal(5, 1))
+        ...    sigma = pyro.sample("x2_sigma", dist.HalfNormal(2))
+        ...    return dist.Normal(mu + parent['x1'], sigma)
+
+        >>> cpd1 = FunctionalCPD("x1", lambda _: x1_prior())
+        >>> cpd2 = FunctionalCPD('x2', fn=lambda parent: x2_prior(parent), parents=['x1'])
+
+        >>> params = model.fit(data, method="MCMC", num_steps=100, mcmc_kwargs={"mp_context": "fork"})
+        >>> print(params["x1_mu"].mean(), params["x1_std"].mean())
+        """
+
+        if not isinstance(data, pd.DataFrame):
+            raise ValueError("Specify data as a pandas Dataframe.")
+        if not isinstance(learning_rate, float):
+            raise ValueError(
+                f"Learning rate should be float type, not {type(learning_rate)}"
+            )
+        if not isinstance(num_steps, int):
+            raise ValueError(
+                f"Number of steps should be int type, not {type(num_steps)}"
+            )
+        if method not in ["SVI", "MCMC"]:
+            raise ValueError("Current implementation only support SVI or MCMC.")
+
+        sort_nodes = list(nx.topological_sort(self))
+
+        if method == "SVI":
+
+            def model():
+                with pyro.plate(f"data", data.shape[0]):
+                    for node in self.nodes:
+                        if node in data.columns:
+                            cpd = self.get_cpds(node)
+                            parents = cpd.parents
+
+                            parent_data = (
+                                {
+                                    p: torch.tensor(data[p].values).float()
+                                    for p in parents
+                                }
+                                if parents
+                                else None
+                            )
+
+                            pyro.sample(
+                                f"{node}",
+                                cpd.fn(parent_data),
+                                obs=torch.tensor(data[node].values),
+                            )
+
+            def guide():
+                # No latent variables to approximate
+                pass
+
+            optimizer = pyro.optim.Adam({"lr": learning_rate})
+            svi = pyro.infer.SVI(
+                model=model,
+                guide=guide,
+                optim=optimizer,
+                loss=pyro.infer.Trace_ELBO(),
+            )
+
+            for step in range(num_steps):
+                loss = svi.step()
+                if step % 50 == 0:
+                    print(f"Step {step} | Loss: {loss:.4f}")
+
+            params = pyro.get_param_store()
+            return {name: params[name].detach().numpy() for name in params.keys()}
+        else:
+            nuts_kwargs = nuts_kwargs or {}
+            mcmc_kwargs = mcmc_kwargs or {}
+
+            def combined_model():
```

Comment:
Can this combined_model be reused in the case of svi and mcmc? So essentially define this combined model first then do an if-else statement for the fitting method.